src/node_modules/@internal/client/node.ts

```diff
@@ -35,30 +42,30 @@ export function configureClient(config: ClientOptions, out: any) {
   out.global[kClient] = createClient({
     ...config,
     exchanges,
-    fetch:
-      config.fetch ||
-      (async (url, options) => {
-        const incomingMessage =
-          (out.stream && (out.stream.req || out.stream.request)) ||
-          out.global.req ||
-          out.global.request;
-        if (typeof url !== "string") throw new Error("Expected a string.");
-        if (!incomingMessage) return fetchImpl(url, options as any) as any;
-        const protocol =
-          incomingMessage.headers["x-forwarded-proto"] ||
-          incomingMessage.protocol;
-        const host =
-          incomingMessage.headers["x-forwarded-host"] ||
-          incomingMessage.headers.host;
-        return fetchImpl(new URL(url, `${protocol}://${host}`).toString(), {
-          ...(options as any),
-          strictSSL,
-          headers: {
-            ...incomingMessage.headers,
-            ...options?.headers,
-          },
-        }) as any;
-      }),
+    fetch: config.fetch
+      ? (...params: Parameters<typeof fetch>) => config.fetch!(...params, fetch)
+      : async (url, options) => {
```

Comment:
@bill-min I think it'd be nice if the fetch passed into the override was the same as the one which is used by default (the async function below) so that it automatically resolves against the current origin, copies headers, etc. On the micro-frame side we're doing similar and just hoist up the common stuff https://github.com/marko-js/micro-frame/blob/main/src/node_modules/%40internal/micro-frame-component/node.marko#L10

Reply:
agree, just pushed another commit.